ca/ca_test.go

```diff
@@ -0,0 +1,187 @@
+package ca
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/bwesterb/mtc"
+)
+
+// testCA is a Merkle Tree CA that runs in memory.
+type testCA struct {
+	*Handle
+	path string
+}
+
+// newTestCA creates a CA in a temporary directory. The caller should call
+// `Close()` in order to delete the directory once the test is finished.
+func newTestCA(t *testing.T) *testCA {
+	path, err := os.MkdirTemp("", "test-mtca-*")
+	if err != nil {
+		t.Fatal(err)
+	}
+
```

Comment:
Add

```go
ok := false
defer func() {
  if !ok {
    os.RemoveAll(ca.path)
  }
}
```

Reply:
I think what you're after is what's implemented by `testCA.Close()` (see below). In fact, we don't want to delete these files until after we've constructed the `testCA` and done the test.

Reply:
For instance, if `issuer.UnmarshalText` fails, then the temporary directory is not removed. Similar if `New` fails.

Reply:
Got it, I see what you're after.

Reply:
Done.